client/client-lib/src/lib.rs

```diff
@@ -551,11 +551,9 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
             self.fetch_all_notes().await;
```

Comment:
I think there is at least another, if not a totally different root cause: `fetch_all_notes` times out now, so if the federation is too slow we don't get our freshly-split e-cash notes and then select a bigger amount.

Reply:
Good point, I'll convert `fetch_all_notes` to return a `Result` to force callers to handle possible failures with it.